soh/soh/Enhancements/randomizer/3drando/spoiler_log.cpp

```diff
@@ -308,25 +308,41 @@ static void WriteLocation(
 
 //Writes a shuffled entrance to the specified node
 static void WriteShuffledEntrance(
-  tinyxml2::XMLElement* parentNode,
-  Entrance* entrance,
-  const bool withPadding = false
-) {
-  auto node = parentNode->InsertNewChildElement("entrance");
-  node->SetAttribute("name", entrance->GetName().c_str());
-  auto text = entrance->GetConnectedRegion()->regionName + " from " + entrance->GetReplacement()->GetParentRegion()->regionName;
-  node->SetText(text.c_str());
-
-  if (withPadding) {
-    constexpr int16_t LONGEST_NAME = 56; //The longest name of a vanilla entrance
-
-    //Insert padding so we get a kind of table in the XML document
-    int16_t requiredPadding = LONGEST_NAME - entrance->GetName().length();
-    if (requiredPadding > 0) {
-      std::string padding(requiredPadding, ' ');
-      node->SetAttribute("_", padding.c_str());
+    Entrance* entrance, const bool withPadding = false) {
+    // Entrance* entrance = Entrances(EntranceKey);
+    int16_t originalIndex = entrance->GetIndex();
+    int16_t destinationIndex = entrance->GetReverse()->GetIndex();
+    int16_t originalBlueWarp = entrance->GetBlueWarp();
+    int16_t replacementIndex = entrance->GetReplacement()->GetIndex();
+    int16_t replacementDestinationIndex = entrance->GetReplacement()->GetReverse()->GetIndex();
+  // auto node = parentNode->InsertNewChildElement("location");
+  switch (gSaveContext.language) {
+        case LANGUAGE_ENG:
+        default:
+            jsonData["Entrances"][std::to_string(originalIndex)] = std::to_string(replacementIndex);
```

Comment:
we're going to want the spoilerfiles to be human readable. using index here will make it so someone reading the spoilerfile has no idea which entrance goes where. it will take a bit more work to parse the spoiler from names than just copying in the index but it will make a huge difference during playtesting

Reply:
no i'm perfectly aware of that, i just want to get it functioning first, then i will fix that.

Reply:
coolcool, just wanted to make sure it was on your radar (it might actually be helpful to do it sooner as opposed to later to help with your own playtesting)

Reply:
will be my first step after getting to function